### PR TITLE
Add comments on CoreLocation authentication requirements

### LINKED ISSUE #2654: Woccatalog on ARM - Geocoding page doesn't work
This seems to be a platform issue (also seeing it on older builds).

### PR DESCRIPTION
**Pull request by aballway**
Fixes #2654

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/2685)

### CODE REVIEW

File: samples/WOCCatalog/WOCCatalog/GeocodingViewController.m [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #import "GeocodingViewController.h"
```
Comment (msft-Jeyaram) [inline samples/WOCCatalog/WOCCatalog/GeocodingViewController.m]: nit: consistency on import vs include

File: include/CoreLocation/CLGeocoder.h [hunk 2]
```diff
@@ -40,4 +40,10 @@ CORELOCATION_EXPORT_CLASS
...
 - (void)cancelGeocode STUB_METHOD;
 @property (readonly, getter=isGeocoding, nonatomic) BOOL geocoding;
-@end
\ No newline at end of file
+@end
+
+@interface CLGeocoder (WinObjC)
+// WINOBJC extension method to set underlying Bing service token, required to use CLGeocoder APIs
+// See https://docs.microsoft.com/en-us/windows/uwp/maps-and-location/geocoding for more info
++ (void)_setServiceToken:(NSString*)token;
```
Comment (DHowett-MSFT) [inline include/CoreLocation/CLGeocoder.h]: nit: underscores are for private things; `setWindowsServiceToken:` might be a good name?

File: samples/WOCCatalog/WOCCatalog/GeocodingViewController.m [hunk 3]
```diff
@@ -65,7 +64,7 @@ - (void)updateTaskInfo {
 
 - (void)startGeocodingCalls {
     // Note: This will not function as the service token is not set
-    // [WSMMapService setServiceToken:@"�"];
+    // [CLGeocoder _setServiceToken:@"�"];
```
Comment (DHowett-MSFT) [inline samples/WOCCatalog/WOCCatalog/GeocodingViewController.m]: gotta fix it here too!

File: Frameworks/CoreLocation/CLGeocoder.mm [hunk 4]
```diff
@@ -113,6 +124,8 @@ void createResultsArray(const MapLocationFinderResult& results, NSMutableArray*
 
 /**
  @Status Interoperable
+ @Notes App must authenticate map service WSMMapService to use (see
```
Comment (DHowett-MSFT) [inline Frameworks/CoreLocation/CLGeocoder.mm]: nit: not totally true.

File: Frameworks/CoreLocation/CLGeocoder.mm [hunk 5]
```diff
@@ -47,6 +47,14 @@ - (instancetype)init {
     return self;
 }
 
+/**
+ @Status Interoperable
+ @Notes Registers app with underlying Bing service (see https://docs.microsoft.com/en-us/windows/uwp/maps-and-location/geocoding)
+*/
++ (void)setWindowsServiceToken:(NSString*)token {
```
Comment (rajsesh-msft) [inline Frameworks/CoreLocation/CLGeocoder.mm]: nit: more of setBingServiceToken or MapsServiceToken, but I digress.

**Pull request merged**